src/main/java/org/hydev/mcpm/client/Downloader.java

```diff
@@ -40,8 +83,45 @@ public void downloadFile(String url, File to)
      */
     public void downloadFiles(Map<String, File> urls, boolean progress, int threads)
     {
-        // TODO: Implement this
-        throw new UnsupportedOperationException("TODO");
+        Thread[] threadss = new Thread[threads];
```

Comment:
I'd try to pick names that are more identifying of what each variable does. In this function you have `t`, `thread`, `threads` and `threadss`... I might consider naming them `currentThreadId`, `threadCount` and `threadList` respectively.